code/modules/mining/mine_structures.dm

```diff
@@ -75,14 +97,49 @@
 		return
 	//Store our light's vars in here
 	flickering = 1
-	spawn(0)
+	spawn()
 		for(var/i = 0; i < amount; i++)
+			if(!lantern)
+				update_brightness()
+				break
 			set_light(0)
-			spawn(rand(5, 15))
-				set_light(6, 2, LIGHT_COLOR_TUNGSTEN)
-		set_light(6, 2, LIGHT_COLOR_TUNGSTEN)
+			sleep(rand(5, 15))
+			update_brightness()
+
 	flickering = 0
```

Comment:
Did you actually steal this part from lights because this is retarded. Flickering would be set to 0 instantly.